distributed/semaphore.py

```diff
@@ -352,28 +344,28 @@ def __init__(
 
         self.refresh_leases = True
 
-        self._registered = None
+        self._do_register = None
         if register:
-            self._registered = self.register()
+            self._do_register = register
 
         # this should give ample time to refresh without introducing another
         # config parameter since this *must* be smaller than the timeout anyhow
-        refresh_leases_interval = (
-            parse_timedelta(
+        lease_timeout = dask.config.get("distributed.scheduler.locks.lease-timeout")
+        if lease_timeout != "inf":
+            lease_timeout = parse_timedelta(
```

Comment:
nit: (less indentation)
```suggestion
        if lease_timeout == "inf":
            return
        lease_timeout = parse_timedelta(
```
Adjust indentation below accordingly, Github doesn't let me create a suggestion for this.